modules/integration/tests/src/test/java/org/wso2/appmanager/integration/utils/HttpUtil.java

```diff
@@ -272,57 +130,86 @@ public static HttpResponse doPost(URL endpoint, String postBody,
-            try {
-                rd = new BufferedReader(new InputStreamReader(
-                        conn.getInputStream()));
-                String line;
-                while ((line = rd.readLine()) != null) {
-                    sb.append(line);
-                }
-                httpResponse = new HttpResponse(sb.toString(), conn.getResponseCode());
-                httpResponse.setResponseMessage(conn.getResponseMessage());
+    private static void closeAll(HttpURLConnection urlConnection, Writer writer, OutputStream out) throws IOException {
+        if (urlConnection != null) {
+            urlConnection.disconnect();
+        }
+        if (out != null) {
+            out.close();
+        }
+        if (writer != null) {
+            writer.close();
+        }
+    }
 
-            } catch (IOException ignored) {
-                rd = new BufferedReader(new InputStreamReader(
-                        conn.getErrorStream()));
-                String line;
-                while ((line = rd.readLine()) != null) {
-                    sb.append(line);
-                }
-                httpResponse = new HttpResponse(sb.toString(), conn.getResponseCode());
-                httpResponse.setResponseMessage(conn.getResponseMessage());
-            } finally {
-                if (rd != null) {
-                    rd.close();
-                }
+    private static void setHeadersToRequest(HttpURLConnection urlConnection, Map<String, String> headers) {
+        if (headers != null && headers.size() > 0) {
+            Iterator<String> itr = headers.keySet().iterator();
+            while (itr.hasNext()) {
+                String key = itr.next();
+                urlConnection.setRequestProperty(key, headers.get(key));
+            }
+        }
+    }
+
+    private static String readResponse(HttpURLConnection urlConnection) throws IOException {
+        InputStream in = null;
+        BufferedReader rd = null;
+        try {
+            StringBuilder sb = new StringBuilder();
+            int responseCode = urlConnection.getResponseCode();
+            if (responseCode / 100 == 2) { //For 20X response codes.
+                in = urlConnection.getInputStream();
+            } else {
+                in = urlConnection.getErrorStream();
+            }
+            rd = new BufferedReader(new InputStreamReader(in));
+            String line;
+            while ((line = rd.readLine()) != null) {
+                sb.append(line);
+            }
+            return sb.toString();
+        } finally {
+            if (rd != null) {
+                rd.close();
+            }
+            if (in != null) {
+                in.close();
             }
-            return httpResponse;
         }
-        return null;
     }
 
+    private static Map<String, String> readHeadersFromResponse(HttpURLConnection urlConnection) {
+        Iterator<String> itr = urlConnection.getHeaderFields().keySet().iterator();
```

Comment:
Fixed in https://github.com/wso2/product-app-manager/pull/357/commits/d5adfb28b2122712d0da0911c0d4921cf5bd98d9